core/evalir.ml

```diff
@@ -238,7 +238,8 @@ struct
         [], Types.Not_typed)) in
     function
     | `FunctionPtr (f, fvs), ps ->
-      let (_finfo, (xs, body), z, _location) = find_fun f in
+      let (_finfo, (xs, body), z, _location) = (try find_fun f with NotFound s ->
```

Comment:
I would suggest putting the try-with on th enext line to make it more obvious there is nontrivial control flow here.

Perhaps also clarify (if I am right) that the change here does not actually contribute to fixing #967, but instead is meant to improve reporting if something goes wrong.  We could even make the debug message a more obvious warning that tells us something is wrong, or translate the NotFound exception to a more informative one that tells the clients of this function that an unsuccessful attempt to look up the function name was made.  (Indeed, a malicious client could try to do this intentionally...)

Reply:
I want to improve reporting if something is not found here. Your suggestion is a good idea to move the debug message to the NotFound exception after making it more informative. I will change the lines accordingly, and will also make the try-with more explicit having new lines here.